sdk/js/packages/client/src/client.ts

```diff
@@ -101,6 +107,61 @@ export class AIChatProtocolClient {
     }
   }
 
+  private prepareRequest(
+    messages: AIChatMessage[],
+    options: AIChatCompletionOperationOptions,
+  ): RequestParameters {
+    if (messages.some((message) => message.files && message.files.length > 0)) {
+      const binaryParts = messages
+        .map((message, index) => {
+          if (message.files) {
+            return message.files.map((file, fileIndex) => ({
+              dispositionType: `form-data; name="messages[${index}].files[${fileIndex}]"`,
+              contentType: file.contentType,
+              body: isUint8Array(file.data)
+                ? createFile(
+                    file.data,
+                    `messages[${index}].files[${fileIndex}]`,
+                  )
+                : file.data,
+            }));
+          }
+          return undefined;
+        })
+        .flat();
+      const boundary = `---Part-${randomUUID()}`;
+      return {
+        contentType: `multipart/form-data; boundary=${boundary}`,
```

Comment:
Is this combo of JSON and image supported across modern browsers?

Reply:
this is just vanilla multipart form data, we just built on top of that.